storybookjs/storybook
# Add `@storybook/nextjs` framework

## What I did

> **Note**: Because our current workflow requires this PR to be merged before we can consistently check it in CI, this work is "feature flagged" to avoid inadvertent use by users of `sb@next init`. See: https://github.com/storybookjs/storybook/pull/19478.

- Add `@storybook/nextjs` framework
- Add nextjs/default-js and nextjs/default-ts to repro configs
- Add basic NEXTJS project type to CLI
    - Includes framework-specific template files
- Update CLI logic
    - Add framework option to baseGenerator
    - Add `hasFrameworkTemplates`
        - Ensures frameworks with templates use those instead of templates for associated renderer
        - Follows pattern of `hasInteractiveStories`
    - Update `copyComponents` to copy common template files before (instead
  of after) framework/render template files
        - Allows framework/renderer template files to contain, e.g. `Introduction.stories.mdx` and have it not be overwritten
- Apply STORYBOOK_REPRO_GENERATOR to env when running `sb init` in repro generator
    - Serves as a "feature flag" around detection of the framework in CLI

> **Note**: This work is _heavily_ based on [storybook-addon-next](https://github.com/RyanClementsHax/storybook-addon-next) by @RyanClementsHax and [storybook-addon-next-router](https://github.com/lifeiscontent/storybook-addon-next-router) by @lifeiscontent. Thank you for your efforts!

## How to test

1. `yarn generate-repros-next --template nextjs/default-js --local-registry`
1. `yarn task --task sandbox --template nextjs/default-js --from-local-repro`
1. Repeat with `nextjs/default-ts`


If your answer is yes to any of these, please make sure to include it in your PR.
